Neos.ContentRepository.Core/Classes/Feature/Common/ContentStreamIdOverride.php

```diff
@@ -24,13 +27,41 @@ class ContentStreamIdOverride
     /**
      * A content stream id that to use instead of the workspace one's {@see ConstraintChecks::requireContentStream()}
      */
-    public static ?ContentStreamId $contentStreamIdToUse = null;
+    private static ?ContentStreamId $contentStreamIdToUse = null;
 
     /**
      * @internal
      */
-    public static function useContentStreamId(?ContentStreamId $contentStreamIdToUse): void
+    public static function withContentStreamIdToUse(ContentStreamId $contentStreamIdToUse, \Closure $fn): void
```

Comment:
"withContentStreamToUse" is a bit bumpy.. I can't think of something better atm.
But more importantly we should add a doc comment explaining this unusual construct even though it is just used internally

Reply:
I would actually avoid `with` as it doesn't follow our usual pattern of how `with` work (aka return a new instance with the property applied).
how about `ContentStreamIdOverride::applyContentStreamId` or even applyContentStreamIdToClosure to be very specific.

Reply:
yeah sounds fine the `with` pattern is actually overloaded if you think of the `withoutAuthentication` function which also accepts a closure.

I sadly havent found yet a good prefix for these kinds of functions that set state and cleanup while another function is called.